packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -3992,4 +3994,79 @@ describe("Teal Opcodes", function () {
       );
     });
   });
+
+  describe("min_balance", () => {
+    useFixture('asa-check');
+    const stack = new Stack<StackElem>();
+
+    // setup 1st account
+    let elon: AccountStoreI = new AccountStore(5e6, { addr: elonAddr, sk: new Uint8Array(0) });
+    setDummyAccInfo(elon);
+
+    // setup 2nd account (to be used as Txn.Accounts[A])
+    const john = new AccountStore(5e6, { addr: johnAddr, sk: new Uint8Array(0) });
+    setDummyAccInfo(john);
+
+    let interpreter: Interpreter;
+    before(() => {
+      interpreter = new Interpreter();
+      interpreter.runtime = new Runtime([elon, john]);
+      interpreter.runtime.ctx.tx.snd = Buffer.from(decodeAddress(elonAddr).publicKey);
+
+      // 'apat': app accounts array
+      interpreter.runtime.ctx.tx.apat = [
+        decodeAddress(johnAddr).publicKey,
+        decodeAddress(generateAccount().addr).publicKey // random account
+      ].map(Buffer.from);
+    });
+
+    it("should push correct account minimum balance", () => {
+      let op = new MinBalance([], 1, interpreter);
+      stack.push(0n); // push sender id
+
+      op.execute(stack);
+      let top = stack.pop();
+      assert.equal(top, BigInt(ALGORAND_ACCOUNT_MIN_BALANCE));
+
+      op = new MinBalance([], 1, interpreter);
+      stack.push(1n); // push sender id
+
+      op.execute(stack);
+      top = stack.pop();
+      assert.equal(top, BigInt(ALGORAND_ACCOUNT_MIN_BALANCE));
+    });
+
+    it("should push raised min_balance to stack after creating asset", () => {
```

Comment:
i think this test should be in flow/integration tests, because here we check operations related to opcode-list only.
min_balance should push `min_balance`. this needs to be tested here

Reply:
> min_balance should push min_balance. this needs to be tested here

did that in previous test

> i think this test should be in flow/integration tests, because here we check operations related to opcode-list only.

in integration tests we test flow in smart contracts. Here we just add an asset and asset that `MinBalance` op should push raised balance.